models/datasets/msf/ds__ncd_msf.sql

```diff
@@ -0,0 +1,267 @@
+    from (
+        select 
+            e.patient_id,
+            e.start_datetime,
+            row_number() over (
+                partition by e.patient_id 
+                order by e.start_datetime desc
+            ) as rn
+        from {{ ref('encounters') }} e
+        join all_ncd_patients anp
+            on e.patient_id = anp.patient_id 
+            and e.start_datetime >= anp.datetime
+        where e.encounter_type = any(array['surveyResponse', 'clinic'])
+    ) recent_encounters
+    where rn <= 38
+    group by patient_id
+),
+
+appointment_history as (
+    select
+        patient_id,
+        array_agg(start_datetime order by start_datetime asc) as appointment_dates
+    from (
+        select 
+            patient_id,
+            start_datetime,
+            row_number() over (
+                partition by patient_id
+                order by start_datetime desc
+            ) as rn
+        from (
+            select distinct
+                e.patient_id,
+                first_value(a.start_datetime) over (
+                    partition by e.patient_id, e.start_datetime
+                    order by a.start_datetime desc
+                ) as start_datetime
+            from {{ ref('outpatient_appointments') }} a
+            join all_ncd_patients anp 
+                on a.patient_id = anp.patient_id 
+                and a.start_datetime >= anp.datetime
+            join (
+                select 
+                    patient_id,
+                    start_datetime,
+                    lead(start_datetime) over (
+                        partition by patient_id 
+                        order by start_datetime
+                    ) as next_encounter_datetime
+                from {{ ref('encounters') }}
+                where encounter_type = any(array['surveyResponse', 'clinic'])
+            ) e
+                on e.patient_id = a.patient_id 
+                and a.start_datetime > e.start_datetime
+                and (
+                    a.start_datetime <= e.next_encounter_datetime
+                    or e.next_encounter_datetime is null
+                )
+        ) last_appointments
+    ) ranked_appointments
+    where rn <= 38
+    group by patient_id
+)
+
+select 
+    p.id as patient_id,
+    p.display_id,
+    concat_ws(' ', nullif(p.first_name, ''), nullif(p.last_name, '')) as patient_name,
+    p.cultural_name,
+    concat_ws(', ',
+        nullif(d.name, ''),
+        nullif(s.name, ''),
+        nullif(t.name, ''),
+        nullif(v.name, ''),
+        nullif(pd.street_village, '')
+    ) as address,
+    date_part('year', age(current_date, p.date_of_birth::date)) as age,
+    p.sex,
+    concat_ws(', ',
+        nullif(pd.primary_contact_number, ''),
+        nullif(pd.secondary_contact_number, '')
+    ) as contact_number,
+    pfv.value as current_status,
+    sv.is_smoker,
+    sv.weight,
+    sv.height,
+    sv.bmi,
+    anp.datetime,
```

Comment:
This should only be populated if the current clinical status is active.
```suggestion
    anp.datetime,
```